service/src/main/java/uk/nhs/adaptors/gp2gp/common/task/TaskHandler.java

```diff
@@ -0,0 +1,45 @@
+package uk.nhs.adaptors.gp2gp.common.task;
+
+import java.util.Optional;
+
+import javax.jms.JMSException;
+import javax.jms.Message;
+
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.common.amqp.JmsReader;
+import uk.nhs.adaptors.gp2gp.common.exception.TaskHandlerException;
+import uk.nhs.adaptors.gp2gp.tasks.TaskDefinition;
+import uk.nhs.adaptors.gp2gp.tasks.TaskDefinitionFactory;
+import uk.nhs.adaptors.gp2gp.tasks.TaskExecutor;
+import uk.nhs.adaptors.gp2gp.tasks.TaskExecutorFactory;
+
+@Component
+@AllArgsConstructor
+@Slf4j
+public class TaskHandler {
+    private static final String MESSAGE_CLASS = "TaskName";
```

Comment:
Variable name doesn't make sense in the context it is used. This is the name of the message header "TaskName" containing the name of the task. That's a lot of names! Maybe consider:

```
...String TASK_TYPE_HEADER_NAME = "TaskType"
...
var taskType = message.getStringProperty(MESSAGE_CLASS);
```